dubbo-registry/dubbo-registry-nacos/src/main/java/org/apache/dubbo/registry/nacos/NacosRegistry.java

```diff
@@ -398,16 +398,17 @@ private void filterServiceNames(Set<String> serviceNames, URL url) {
 
     private void subscribeEventListener(String serviceName, final URL url, final NotifyListener listener)
             throws NacosException {
-        if (!nacosListeners.containsKey(serviceName)) {
-            EventListener eventListener = event -> {
-                if (event instanceof NamingEvent) {
-                    NamingEvent e = (NamingEvent) event;
-                    notifySubscriber(url, listener, e.getInstances());
-                }
-            };
-            namingService.subscribe(serviceName, eventListener);
-            nacosListeners.put(serviceName, eventListener);
+        if (nacosListeners.containsKey(serviceName)) {
+            logger.info("nacosListeners contains serviceName:" + serviceName);
         }
+        EventListener eventListener = event -> {
+            if (event instanceof NamingEvent) {
+                NamingEvent e = (NamingEvent) event;
+                notifySubscriber(url, listener, e.getInstances());
+            }
+        };
+        namingService.subscribe(serviceName, eventListener);
+        nacosListeners.put(serviceName, eventListener);
```

Comment:
I think we should also consider changing the data structure of nacosListeners to Map<String, List<EventListener>>?